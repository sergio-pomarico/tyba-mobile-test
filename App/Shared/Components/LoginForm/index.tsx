import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { Field, reduxForm } from 'redux-form';

import Input from '../Input';
import Button from '../Button';

interface LoginFormProps {
  handleSubmit: (values: any) => void;
  handleLogin: (email: string, password: string) => void;
}

interface Values {
  email: string;
  password: string;
}

class LoginForm extends Component<LoginFormProps> {
  render() {
    const { handleSubmit, handleLogin } = this.props;
    return (
      <View style={styles.container}>
        <Field name="email" component={Input} placeholder="email" isEmail />
        <Field
          name="password"
          component={Input}
          placeholder="password"
          isPass
        />
        <Button
          text="Login"
          onPress={handleSubmit(({ email, password }) =>
            handleLogin(email, password)
          )}
        />
      </View>
    );
  }
}

const validate = (values: Values) => {
  const errors: any = {};

  if (!values.email) {
    errors.email = 'email is required';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'invalid email address';
  }

  if (!values.password) {
    errors.password = 'password is required';
  } else if (values.password.length < 6) {
    errors.password = 'password must have at least 6 characters';
  }

  return errors;
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%'
  }
});

export default reduxForm({ form: 'LoginForm', validate })(LoginForm);
