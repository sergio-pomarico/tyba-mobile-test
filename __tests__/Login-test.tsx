
import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import LoginForm from '../App/Shared/Components/LoginForm';

configure({adapter: new Adapter()});
const middlewares: Array<any> = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);
const initialState = {};

describe('Testing LoginForm', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <LoginForm />,
      { context: { store: mockStore(initialState) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
