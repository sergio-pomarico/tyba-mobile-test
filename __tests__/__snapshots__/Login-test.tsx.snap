// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginForm renders as expected 1`] = `ShallowWrapper {}`;
